esp/services/ws_sql/ws_sqlService.cpp

```diff
@@ -1179,10 +1179,31 @@ void CwssqlEx::createWUXMLParams(StringBuffer & xmlparams, HPCCSQLTreeWalker* pa
                 xmlparams.append("<").append(varname.s.str()).append(">");
 
                 IConstNamedValue &item = variables->item(i);
-                const char *value = item.getValue();
-                if(value && *value)
-                  encodeXML(value, xmlparams);
+
+                char * value = ((char *)item.getValue());
+
+                if (value && *value)
+                {
+                    while(value && isspace(*value)) //fast trim left
+                        value++;
+
+                    int len = strlen(value);
+
+                    while(len && isspace(value[len-1])) //fast trim right
+                        value[--len] = '\0';
```

Comment:
Consider remembering the offset of most recently found (leftmost) space, and after you exit the loop set it to NULL